frontend/src/metabase/visualizations/echarts/cartesian/waterfall/option.ts

```diff
@@ -20,13 +36,35 @@ export function getWaterfallOption(
     renderingContext,
   );
 
-  // TODO remove typecast
+  // dataset
   (option.dataset as DatasetOption[])[0].dimensions =
     Object.values(DATASET_DIMENSIONS);
-  // TODO full yAxis options
-  option.yAxis = {
-    type: "value",
-  };
+
+  // x-axis
+  (option.xAxis as XAXisOption).type = getXAxisType(settings);
+
+  // y-axis
+  const extent = getWaterfallExtent(chartModel.dataset as WaterfallDataset);
+  chartModel.yAxisExtents = [extent, null];
+
+  const axisRanges = getAxisRanges(chartModel, settings);
+  const { left: leftAxisFormatter } = getAxesFormatters(
```

Comment:
Heads up, I slightly changed it and there is the axis model in the chart model that has the formatting function https://github.com/metabase/metabase/pull/36879